controllers/azurecluster_controller.go

```diff
@@ -221,9 +222,18 @@ func (acr *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	}
 
 	if err := acs.Reconcile(ctx); err != nil {
-		// Handle transient errors
+		// Handle terminal and transient errors
 		var reconcileError azure.ReconcileError
 		if errors.As(err, &reconcileError) {
+			if reconcileError.IsTerminal() {
```

Comment:
a couple of build errors still:

```
controllers/azurecluster_controller.go:230:76: clusterScope.Name undefined (type *scope.ClusterScope has no field or method Name)
controllers/azurecluster_controller.go:231:17: clusterScope.SetFailureReason undefined (type *scope.ClusterScope has no field or method SetFailureReason)
controllers/azurecluster_controller.go:232:17: clusterScope.SetFailureMessage undefined (type *scope.ClusterScope has no field or method SetFailureMessage)
controllers/azurecluster_controller.go:233:17: clusterScope.SetNotReady undefined (type *scope.ClusterScope has no field or method SetNotReady)
controllers/azurecluster_controller.go:234:17: clusterScope.SetVMState undefined (type *scope.ClusterScope has no field or method SetVMState)
```

Reply:
instead of `Name()` you can use `ClusterName`

Reply:
AzureClusters also don't have a `FailureMessage` or `SetFailureReason` in status so line 231 and 232 can be removed

Reply:
ok